pilot/pkg/features/experimental.go

```diff
@@ -190,10 +190,6 @@ var (
 	PassthroughTargetPort = env.Register("ENABLE_RESOLUTION_NONE_TARGET_PORT", true,
 		"If enabled, targetPort will be supported for resolution=NONE ServiceEntry").Get()
 
-	PersistOldestWinsHeuristicForVirtualServiceHostMatching = env.Register("PERSIST_OLDEST_FIRST_HEURISTIC_FOR_VIRTUAL_SERVICE_HOST_MATCHING", false,
```

Comment:
I'm fairly concerned about removing the ability to persist the old behavior. We've given users less than a year to refactor the way their VirtualServices have worked for the last 3-4 years. Add to that a lack of publicity about these sorts of behavioral changes and this feels like pain waiting to happen

Reply:
Here is my rationale. Consider a user on 1.19 that cares about this behavior.

1. They upgrade to 1.20 and decide they need this flag, so they enable it.
2. Now 12 months pass and 1.24 comes out and they see the flag is removed. At this point they had 12 months to do any changes required. Ideally, this is done and they need no action...
3. If not, they now are blocked on upgrading. They have another 9 months before their 1.23 version is EOL

So overall, they get 12 months "free", and 9 months CVE free, for a total of up to 21 months to migrate. This is pretty generous. Historically, we have found that if a user is not going to make a change in <some reasonable period of time>, giving them more time doesn't actually make things better -- they will just only make the change when they hit the hard deadline.

Using experimental **temporary** feature flags is a shared responsibility model between users and maintainers, where users are expected to migrate off of them and provide us feedback on them. At this point, there has not been a single user who has mentioned even needing to set this flag.

In the past, we have gotten feedback from customers are *very* conservative with upgrades that we removed flags too quickly. What they wanted was to have 4 releases of support -- which is what we are doing here.

So its not clear to me really who we are helping by keeping these around longer. 

We also document 'Currently, each compatibility version is intended to remain for at least two versions, though this is subject to change.'; we are already supporting beyond that.

Reply:
Fair point, removing the hold